optimizer/plan/planbuilder.go

```diff
@@ -203,3 +213,65 @@ func (b *planBuilder) buildPrepare(x *ast.PrepareStmt) Plan {
 	}
 	return p
 }
+
+func (b *planBuilder) buildAdmin(adm *ast.AdminStmt) Plan {
+	if adm.Tp == ast.AdminCheckTable {
+		if adm.Tables != nil {
+			return &CheckTable{Tables: adm.Tables}
+		}
+
+		return nil
+	}
+
+	if adm.Tp == ast.AdminShowDDL {
+		p := &ShowDDL{}
+		p.SetFields(buildShowDDLFields())
+
+		return p
+	}
+
+	return nil
+}
+
+func buildShowDDLFields() []*ast.ResultField {
+	tbName := tableShowDDL
```

Comment:
We should not give it a table name, it's not a real table.
tbName should be "".